mne/time_frequency/tests/test_spectrum.py

```diff
@@ -219,3 +222,50 @@ def test_spectrum_proj(inst, request):
     with has_proj.info._unlock():
         has_proj.info['projs'] = no_proj.info['projs']
     assert has_proj == no_proj
+
+
+@pytest.mark.parametrize('method, average', [
+    ('welch', False),
+    ('multitaper', True),
+    ('multitaper', False),
+])
+def test_spectrum_complex(method, average):
+    """Test output='complex' support."""
+    sfreq = 100
+    n = 10 * sfreq
+    freq = 3.
+    phase = np.pi / 4  # should be recoverable
+    data = np.cos(2 * np.pi * freq * np.arange(n) / sfreq + phase)[np.newaxis]
+    raw = RawArray(data, create_info(1, sfreq, 'eeg'))
+    epochs = make_fixed_length_epochs(raw, duration=2., preload=True)
+    assert len(epochs) == 5
+    assert len(epochs.times) == 2 * sfreq
+    kwargs = dict(output='complex', method=method)
+    if method == 'welch':
+        kwargs['n_fft'] = sfreq
+        ctx = pytest.warns(UserWarning, match='Zero value')
+        want_dims = ('epoch', 'channel', 'freq')
+        want_shape = (5, 1, sfreq // 2 + 1)
+        if not average:
+            want_dims = want_dims + ('segment',)
+            want_shape = want_shape + (2,)
+            kwargs['average'] = average
```

Comment:
looking at the `parametrize` it appears you're not testing `method='welch', average=True` (for obvious reasons) which makes me wonder why bother with the `if` clause. It doesn't hurt anything though, so 🤷🏻

Reply:
I'll add it. In cases where you end up averaging anyway (like I do!) because you've ensured your epochs are in-phase with the freq of interest it should be equivalent anyway